akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1162,12 +1164,38 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    */
   // FIXME: I don't like the Pair allocation :(
   @ApiMayChange
-  final protected def getStageActor(receive: ((ActorRef, Any)) ⇒ Unit): StageActor = {
+  final protected def getStageActor(receive: ((ActorRef, Any)) ⇒ Unit): StageActor =
+    getStageActor(receive, name = "")
+
+  /**
+   * Initialize a named [[StageActorRef]] which can be used to interact with from the outside world "as-if" an [[Actor]].
+   * The messages are looped through the [[getAsyncCallback]] mechanism of [[GraphStage]] so they are safe to modify
+   * internal state of this stage.
+   *
+   * This method must (the earliest) be called after the [[GraphStageLogic]] constructor has finished running,
+   * for example from the [[preStart]] callback the graph stage logic provides.
+   *
+   * Created [[StageActorRef]] to get messages and watch other actors in synchronous way.
+   *
+   * The [[StageActorRef]]'s lifecycle is bound to the Stage, in other words when the Stage is finished,
+   * the Actor will be terminated as well. The entity backing the [[StageActorRef]] is not a real Actor,
+   * but the [[GraphStageLogic]] itself, therefore it does not react to [[PoisonPill]].
+   *
+   * @param receive callback that will be called upon receiving of a message by this special Actor
+   * @param name to be used in the Actor's name. An empty String will cause the usual auto generated name to be used
+   * @return minimal actor with watch method
+   */
+  @ApiMayChange
+  final protected def getStageActor(receive: ((ActorRef, Any)) ⇒ Unit, name: String): StageActor = {
```

Comment:
StageActors can now have names (the name is added as part of the name, it still contains the functionref part being `$$random`